dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,40 +53,176 @@ public Pattern getParameterPattern() {
+        }
+    }
+
+    private class DatabaseTable extends DatabaseObject {
+
+        private DatabaseTable(Connection conn, String defaultSchemaName, String objName) {
+            super(conn, defaultSchemaName, objName);
+        }
+
+        @Override
+        public boolean exists() throws SQLException {
+            ResultSet rs = con.getMetaData().getTables(null, name.getSchemaName(), name.getObjectName(), null);
+            return rs.next();
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return con.getMetaData().getColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs) {
+            return Direction.INPUT;
+        }
+    }
+
+    private class DatabaseProcedure extends DatabaseObject {
+
+        private DatabaseProcedure(Connection conn, String defaultSchemaName, String objName) {
+            super(conn, defaultSchemaName, objName);
+        }
+
+        @Override
+        public boolean exists() throws SQLException {
+            ResultSet rs = con.getMetaData().getProcedures(null, name.getSchemaName(), name.getObjectName());
+            return rs.next();
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return con.getMetaData().getProcedureColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs)
+                                        throws SQLException {
+            Direction dir = null;
+            switch (rs.getInt("COLUMN_TYPE")) {
+                case DatabaseMetaData.procedureColumnIn:
+                    dir = Direction.INPUT;
+                    break;
+                case DatabaseMetaData.procedureColumnInOut:
+                    dir = Direction.INPUT_OUTPUT;
+                    break;
+                case DatabaseMetaData.procedureColumnOut:
+                    dir = Direction.OUTPUT;
+                    break;
+            }
+            return dir;
+        }
+    }
+
+    private class DatabaseFunction extends DatabaseObject {
+
+        private DatabaseFunction(Connection conn, String defaultSchemaName, String objName) {
+            super(conn, defaultSchemaName, objName);
+        }
+
+        @Override
+        public boolean exists() throws SQLException {
+            ResultSet rs = con.getMetaData().getFunctions(null, name.getSchemaName(), name.getObjectName());
+            return rs.next();
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return con.getMetaData().getFunctionColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs)
+                                        throws SQLException {
```

Comment:
I think the method name and throws clause is short enough to fit in single line here.